cluster/store/schema.go

```diff
@@ -303,15 +303,30 @@ func (s *schema) getTenants(class string) ([]*models.Tenant, error) {
 	// Read tenants using the meta lock guard
 	var res []*models.Tenant
 	f := func(_ *models.Class, ss *sharding.State) error {
-		res = make([]*models.Tenant, len(ss.Physical))
+		if limit != nil {
+			res = make([]*models.Tenant, *limit)
+		} else {
+			res = make([]*models.Tenant, len(ss.Physical))
+		}
+		foundAfter := after == nil
 		i := 0
 		for tenant := range ss.Physical {
+			if !foundAfter {
```

Comment:
sadly this will not work reliably as `ss.Physical` is a map and the order is not guranteed to be the same - AFAIK you would get a different order after a restart which could result in incomplete iterations

Reply:
Ahh yes you are right, totally forgot about that (haven't been on Go land in a bit; stuck in React at work the past year). Let me circle back and see what I can do.

Thanks for taking a look so quickly @dirkkul

Reply:
@dirkkul I'm not at my computer so I can't really dive into the code well, but will those suffice?: just iterate through the physical store map and order the tenants and proceed? Are tenants named in an ordinal manner?

Reply:
> just iterate through the physical store map and order the tenants and proceed?

The problem is that there are setup with possibly millions of tenants => we'd need to allocate space to fit all of them, which might just be too much.

Reply:
that's what I was worried about. I guess that voids this issue/feature request 😅.